aries/aries_vcx_core/src/wallet2/mod.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use derive_builder::Builder;
+#[cfg(feature = "vdrtools_wallet")]
+use indy_api_types::domain::wallet::Record as IndyRecord;
+use serde::{Deserialize, Serialize};
+
+use self::entry_tag::{EntryTag, EntryTags};
+use crate::{errors::error::VcxCoreResult, wallet::structs_io::UnpackMessageOutput};
+
+#[cfg(feature = "vdrtools_wallet")]
+pub mod indy_wallet;
+
+pub mod entry_tag;
+
+pub struct Key {
+    pub pubkey_bs58: String,
+}
+
+#[derive(Debug, Default, Clone, Builder)]
+pub struct Record {
+    pub category: String,
+    pub name: String,
+    pub value: String,
+    #[builder(default = "EntryTags::default()")]
+    pub tags: EntryTags,
+}
+
+#[derive(Debug, Default, Clone, Builder)]
+pub struct RecordUpdate {
+    pub category: String,
+    pub name: String,
+    #[builder(setter(strip_option), default = "None")]
+    pub value: Option<String>,
+    #[builder(setter(strip_option), default = "None")]
+    pub tags: Option<EntryTags>,
+}
+
+#[cfg(feature = "vdrtools_wallet")]
+impl From<IndyRecord> for Record {
+    fn from(ir: IndyRecord) -> Self {
+        let tags = ir
+            .tags
+            .into_iter()
+            .map(|(key, value)| EntryTag::Plaintext(key, value))
+            .collect();
+        Self {
+            name: ir.id,
+            category: ir.type_,
+            value: ir.value,
+            tags,
+        }
+    }
+}
+
+#[cfg(feature = "vdrtools_wallet")]
+impl From<Record> for IndyRecord {
+    fn from(record: Record) -> Self {
+        let tags = record
+            .tags
+            .into_iter()
+            .fold(HashMap::new(), |mut memo, item| {
+                match item {
+                    EntryTag::Encrypted(key, val) => memo.insert(key, val),
+                    EntryTag::Plaintext(key, val) => memo.insert(format!("~{}", key), val),
+                };
+                memo
+            });
+        Self {
+            id: record.name,
+            type_: record.category,
+            value: record.value,
+            tags,
+        }
+    }
+}
```

Comment:
It seems that this conversion is never used, I tried to delete it, both this and #1093 still work without it

Reply:
Correct, not needed.